File: src/main/java/zemoa/StartAppContext.java
package zemoa;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.ApplicationListener;
import org.springframework.stereotype.Component;

@Component
public class StartAppContext implements ApplicationListener<AppStartingEvent> {
    private static final Logger LOGGER = LoggerFactory.getLogger(StartAppContext.class);
    @Override
    public void onApplicationEvent(AppStartingEvent appStartingEvent) {
        LOGGER.warn("App Starting");
    }
}

File: src/main/java/zemoa/StartAppContext.kt
package zemoa

import org.slf4j.LoggerFactory
import org.springframework.context.ApplicationListener
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.stereotype.Component
import zemoa.StartAppContext

@EnableAsync
@Component
class StartAppContext : ApplicationListener<AppStartingEvent> {
    companion object {
        private val LOGGER = LoggerFactory.getLogger(StartAppContext::class.java)
    }


    override fun onApplicationEvent(event: AppStartingEvent) {
        LOGGER.info("App Starting using java version ${SystemInfo.javaVersion()} and javafx version ${SystemInfo.javafxVersion()}")
    }
}
